src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/LoggingExtensions.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Threading.Tasks
+{
+    internal static class LoggingExtensions
+    {
+        public static string GetMethodName(this Delegate @delegate)
+        {
+            DiagnosticMethodInfo? dmi = DiagnosticMethodInfo.Create(@delegate);
+            return dmi?.Name ?? "<unknown>";
```

Comment:
This could alternatively be this, but not sure how important that is:

```suggestion
            if (StackTrace.IsSupported)
            {
                DiagnosticMethodInfo? dmi = DiagnosticMethodInfo.Create(@delegate);
                return dmi!.Name;
            }
            else
            {
                return @delegate.Method.Name;
            }
```